src/server/game/Spells/SpellEffects.cpp

```diff
@@ -929,8 +929,14 @@ void Spell::EffectTeleportUnits(SpellEffIndex /*effIndex*/)
         orientation = m_targets.GetUnitTarget()->GetOrientation();
     TC_LOG_DEBUG("spells", "Spell::EffectTeleportUnits - teleport unit to %u %f %f %f %f\n", mapid, x, y, z, orientation);
 
-    if (unitTarget->GetTypeId() == TYPEID_PLAYER)
-        unitTarget->ToPlayer()->TeleportTo(mapid, x, y, z, orientation, unitTarget == m_caster ? TELE_TO_SPELL | TELE_TO_NOT_LEAVE_COMBAT : 0);
+    if (Player* player = unitTarget->ToPlayer())
+    {
+        // Custom loading screen
+        if (uint32 customLoadingScreenId = m_spellInfo->GetEffect(effIndex)->MiscValue)
```

Comment:
Thanks,
I guess my intellisense was slow, I expected effectInfo as well but went ahead and used m_spellInfo because I got no response.
Resharper's intellisense problem is starting to get a bit annoying.